clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -17192,8 +17185,8 @@ Decl *Sema::BuildStaticAssertDeclaration(SourceLocation StaticAssertLoc,
             << InnerCond->getSourceRange();
         DiagnoseStaticAssertDetails(InnerCond);
       } else {
-        Diag(AssertExpr->getBeginLoc(), diag::err_static_assert_failed)
-            << !HasMessage << Msg.str() << AssertExpr->getSourceRange();
+        Diag(StaticAssertLoc, diag::err_static_assert_failed)
+            << !AssertMessage << Msg.str() << AssertExpr->getSourceRange();
```

Comment:
This two lines are undoing the changes in:

- 66202d83b5d47 (https://reviews.llvm.org/D147745) on 2023-04-13, which modifies the code around the one the originally commit also modified before.
- 47ccfd7a89e2a (https://reviews.llvm.org/D154290) on 2023-07-19, which modifies the code introduced by the original commit.